execution_workflows/APA-Scan/workflow/Snakefile

```diff
@@ -0,0 +1,337 @@
+# So I imagine this'll carry on from the previous RNA-Seq alignment step. They just
+# need to be in bam format and there can be multiple in a directory.
+# MAY NEED TO ADJUST THE INDEXING DEPENDING ON THE LEVELS, BUT CAN PROBABLY MAKE
+# THIS CUSTOMISABLE FROM THE COMMAND LINE OR A CONFIG FILE ETC. THIS IS BECAUSE
+# THE LAYOUT OF THESE WILL DICTATE THE ORIENTATION OF LEVELS THAT ARE COMPARED
+# BY APA-SCAN. IE FOR THE TEST DATA, THIS MAKES SO SRSF3 is CONDITION1 AND
+# THE CONTROL IS CONDITION 2
+
+################################################################################
+
+
+
+# Parameters
+THREADS = config["threads"]
+
+# Conda
+CONDA_YAML = config["method"] + ".yaml"
+
+# Scripts
+CONVERT_ID_PYSCRIPT = config["scripts"] + "/convert_geneids.py"
+
+# Input file directories
+CURRENT_DIR = os.getcwd()
+CONDITION1 = samples["sample_type"].unique()[0]
+CONDITION2 = samples["sample_type"].unique()[1]
+
+# Output filename shorthands
+PARAMCODE = config["paramcode"]
+APA_SCAN_OUTPUT = config["method"].replace("-","_") + "_" + CONDITION1 + "_Vs_" + CONDITION2 + ".xlsx" # Raw output of APA-Scan.py
+APA_SCAN_OUTPREFIX = PARAMCODE + "_" + config["method"] # Prefix for final APA-Scan output
+
+# Annotation file
+# These NEED to be in the same directory you run APA-Scan.py from.
+ANNOTATION = config["annotation"]
+GENOMEFILE = config["genome"]
+
+
+# Print to compare conditions and check correct way round
+print(">>>>> COMPARING: ")
+print("   >> " + CONDITION1)
+print("   >> " + CONDITION2)
+print("   >> APA-Scan RAW OUTPUT IS :- " + config["out_dir"] + "/" + APA_SCAN_OUTPUT)
+
+
+
+
+
+# Final Target Rule ------------------------------------------------------------
+localrules: finish
+
+# Final output .xlsx file, that's been renamed with rename_output.
+
+rule finish:
+    """Rule that specifies the final output.
+    """
+    input:
+        os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_01.bed"),
+        os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_03.tsv")
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Preprocessing: obtain suitable input formats
+
+
+#### NOTES #####################################################################
+
+# What's going on here is that APA-Scan is set up so it reads a specific file,
+# "configuration.ini" to get the information it needs to run, such as where
+# the data, annotation and genome files are and how to output them.
+# What this function below does is there's a "blank" configuration file present
+# which has fields that correspond to different directories and files etc. These
+# are in the form "dataofinterest_dir", for example input_dir1, input_dir2,
+# annotation_dir and genome_dir. Using sed, we replace these tags with our
+# dirs/files of interest, which are set at the top. Output is just set to the
+# curren working directory as APA-Scan again is really painful and picky so
+# best not to tempt fate and the bioinformatics demons.
```

Comment:
I'd move this to the comment string (line 208/209) under the rule (and trim it a little bit, but my hypocrite sensor is going crazy right now :))